### PR TITLE
MINOR: code cleanup

### PR DESCRIPTION
**Pull request by mjsax**
Similar PR as #6053

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @guozhangwang @bbejeck @vvcephei

**Review by bbejeck — APPROVED** [review]

**Comment by bbejeck** [comment]
retest this please

**Comment by mjsax** [comment]
Retest this please.

File: streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java [hunk 1]
```diff
@@ -64,84 +64,143 @@ public void shouldNotThrowNullPointerIfOptimizationsNotSpecified() {
...
+            .join(filteredKTable, MockValueJoiner.TOSTRING_JOINER);
         builder.build();
 
-        final ProcessorTopology topology = builder.internalTopologyBuilder.rewriteTopology(new StreamsConfig(props)).build();
-
-        assertThat(topology.stateStores().size(), equalTo(1));
-        assertThat(topology.processorConnectedStateStores("KSTREAM-JOIN-0000000005"), equalTo(Collections.singleton(topology.stateStores().get(0).name())));
-        assertThat(topology.processorConnectedStateStores("KTABLE-FILTER-0000000003").isEmpty(), is(true));
+        final ProcessorTopology topology
+            = builder.internalTopologyBuilder.rewriteTopology(new StreamsConfig(props)).build();
```
Comment (vvcephei) [inline streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java]: Continuing from https://github.com/apache/kafka/pull/6053#discussion_r244879693, it seems strange to place the assignment at the beginning of a line.

This pattern recurs a lot in this PR, so I'll not comment on it further.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/CachingSessionStoreTest.java [hunk 2]
```diff
@@ -154,7 +152,8 @@ public void shouldFlushItemsToStoreOnEviction() {
         final StateSerdes<Bytes, byte[]> serdes = new StateSerdes<>("topic", Serdes.Bytes(), Serdes.ByteArray());
         final List<KeyValue<Windowed<Bytes>, byte[]>> added = addSessionsUntilOverflow("a", "b", "c", "d");
         assertEquals(added.size() - 1, cache.size());
-        final KeyValueIterator<Windowed<Bytes>, byte[]> iterator = WrappedSessionStoreIterator.bytesIterator(underlying.fetch(added.get(0).key.key(), 0, 0), serdes);
+        final KeyValueIterator<Windowed<Bytes>, byte[]> iterator
+            = WrappedSessionStoreIterator.bytesIterator(underlying.fetch(added.get(0).key.key(), 0, 0), serdes);
```
Comment (vvcephei) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/CachingSessionStoreTest.java]: Could shorten this line with a static import and/or splitting it.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/CachingWindowStoreTest.java [hunk 3]
```diff
@@ -83,7 +82,8 @@
     public void setUp() {
         keySchema = new WindowKeySchema();
         underlying = new RocksDBSegmentedBytesStore("test", "metrics-scope", 0, SEGMENT_INTERVAL, keySchema);
-        final RocksDBWindowStore<Bytes, byte[]> windowStore = new RocksDBWindowStore<>(underlying, Serdes.Bytes(), Serdes.ByteArray(), false, WINDOW_SIZE);
+        final RocksDBWindowStore<Bytes, byte[]> windowStore
+            = new RocksDBWindowStore<>(underlying, Serdes.Bytes(), Serdes.ByteArray(), false, WINDOW_SIZE);
```
Comment (vvcephei) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/CachingWindowStoreTest.java]: This line is still long.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/CachingWindowStoreTest.java [hunk 4]
```diff
@@ -169,7 +167,8 @@ public void close() {
         final long initialWallClockTime = 0L;
         final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), streamsConfiguration, initialWallClockTime);
 
-        final ConsumerRecordFactory<String, String> recordFactory = new ConsumerRecordFactory<>(Serdes.String().serializer(), Serdes.String().serializer(), initialWallClockTime);
+        final ConsumerRecordFactory<String, String> recordFactory
+            = new ConsumerRecordFactory<>(Serdes.String().serializer(), Serdes.String().serializer(), initialWallClockTime);
```
Comment (vvcephei) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/CachingWindowStoreTest.java]: also still long

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java [hunk 5]
```diff
@@ -225,6 +304,17 @@ public void testRangeAndSinglePointFetch() {
         assertNull(entriesByKey.get(6));
     }
 
+    private void verifyStore(final String expectedValue,
```
Comment (vvcephei) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: It looks like this method has just one usage. Should it be inlined?

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java [hunk 6]
```diff
@@ -573,8 +839,203 @@ public void testRolling() {
 
     }
 
+    private void verifyStoreContent1(final long startTime,
```
Comment (vvcephei) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: Same question about this and the following methods... It looks like they're only used once. Should we inline them?

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java [hunk 7]
```diff
@@ -605,36 +1066,138 @@ public void testRestore() throws IOException {
...
-            Utils.mkSet(segments.segmentName(4L), segments.segmentName(5L), segments.segmentName(6L)),
+            Utils.mkSet(
+                segments.segmentName(4L),
+                segments.segmentName(5L),
+                segments.segmentName(6L)),
             segmentDirs(baseDir)
         );
     }
 
+    private void verifyEmpty(final long startTime,
```
Comment (vvcephei) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: Ditto on this method, which also appears to be used just once.

File: streams/src/test/java/org/apache/kafka/test/MockProcessor.java [hunk 8]
```diff
@@ -16,17 +16,17 @@
...
 import org.apache.kafka.streams.processor.ProcessorContext;
 import org.apache.kafka.streams.processor.PunctuationType;
-import org.apache.kafka.streams.processor.Punctuator;
 
+import java.time.Duration;
 import java.util.ArrayList;
 
 import static org.junit.Assert.assertEquals;
 
+@SuppressWarnings("WeakerAccess")
```
Comment (vvcephei) [inline streams/src/test/java/org/apache/kafka/test/MockProcessor.java:29]: Just curious, why is this necessary?

**Review by vvcephei — COMMENTED** [review]
LGTM. Thanks for this cleanup also. Just some nits...

File: streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java [hunk 9]
```diff
@@ -603,7 +588,8 @@ public void statelessTopologyShouldNotCreateStateDirectory() throws Exception {
         topology.addSource("source", Serdes.String().deserializer(), Serdes.String().deserializer(), inputTopic)
                 .addProcessor("process", () -> new AbstractProcessor<String, String>() {
                     @Override
-                    public void process(final String key, final String value) {
+                    public void process(final String key,
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java]: nit: I think this does not need to be broken to two lines as it is still quite short, and less than 3 parameters. Ditto below.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java [hunk 10]
```diff
@@ -87,11 +87,14 @@ public void shouldRespectBulkloadOptionsDuringInit() {
         restoreListener.onRestoreStart(null, rocksDBStore.name(), 0L, 0L);
 
         assertThat(rocksDBStore.getOptions().level0FileNumCompactionTrigger(), equalTo(1 << 30));
+        assertThat(rocksDBStore.getOptions().level0SlowdownWritesTrigger(), equalTo(1 << 30));
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java:90]: These are beyond code cleanup, wondering if it is related to jenkins failures?

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java (hunk 5 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: +1

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java (hunk 6 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: I'm a bit skeptical for functions whose name need a number suffix to distinguish with others, since it usually indicates loss of generality :P Could the common logic be shared, or if not I'd suggest we still inline all of them.

File: streams/src/test/java/org/apache/kafka/test/MockProcessor.java [hunk 11]
```diff
@@ -67,13 +69,13 @@ public void punctuate(final long timestamp) {
...
                             .add(timestamp);
-                }
-            });
+                });
         }
     }
 
     @Override
-    public void process(final K key, final V value) {
+    public void process(final K key,
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/test/MockProcessor.java]: ditto here and below

**Review by guozhangwang — COMMENTED** [review]
Made a pass.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/CachingSessionStoreTest.java [hunk 12]
```diff
@@ -163,7 +162,8 @@ public void shouldFlushItemsToStoreOnEviction() {
     @Test
     public void shouldQueryItemsInCacheAndStore() {
         final List<KeyValue<Windowed<Bytes>, byte[]>> added = addSessionsUntilOverflow("a");
-        final KeyValueIterator<Windowed<Bytes>, byte[]> iterator = cachingStore.findSessions(Bytes.wrap("a".getBytes()), 0, added.size() * 10);
+        final KeyValueIterator<Windowed<Bytes>, byte[]> iterator
+            = cachingStore.findSessions(Bytes.wrap("a".getBytes()), 0, added.size() * 10);
```
Comment (ryannedolan) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/CachingSessionStoreTest.java]: maybe worth specifying charset here

File: streams/src/test/java/org/apache/kafka/streams/state/internals/CachingWindowStoreTest.java [hunk 13]
```diff
@@ -288,7 +311,8 @@ public void shouldFetchAllWithinTimestampRange() {
     public void shouldFlushEvictedItemsIntoUnderlyingStore() {
         final int added = addItemsToCache();
         // all dirty entries should have been flushed
-        final KeyValueIterator<Bytes, byte[]> iter = underlying.fetch(Bytes.wrap("0".getBytes()), DEFAULT_TIMESTAMP, DEFAULT_TIMESTAMP);
+        final KeyValueIterator<Bytes, byte[]> iter
+            = underlying.fetch(Bytes.wrap("0".getBytes()), DEFAULT_TIMESTAMP, DEFAULT_TIMESTAMP);
```
Comment (ryannedolan) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/CachingWindowStoreTest.java]: same

File: streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java (hunk 9 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java]: I like it better if we always break it down. But I will revert if you prefer.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java (hunk 10 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java:90]: Jenkins failed in core test (not sure about older runs -- build was deleted already).

This just closes a test gap.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java (hunk 5 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: Because of new line breaks, the method got too long, and thus, I extracted some parts into own methods to make the build pass (otherwise checkstyle fails).

Happy to address the issue otherwise. But inlining won't work. Please let me know what you think.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java (hunk 10 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java:90]: Ack, thanks for confirming.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java (hunk 6 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: See comment above. The methods have nothing in common besides that they are verifying the store content at some point during the test. Not happy with the naming either, but had no better idea.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java (hunk 7 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: compare comment above

File: streams/src/test/java/org/apache/kafka/test/MockProcessor.java (hunk 8 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/test/MockProcessor.java:29]: Intellij warns that some member (eg, `processedKeys`) are public.

File: streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java (hunk 1 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java]: Did a search with Intellij. Hope I found all of them :)

**Comment by mjsax** [comment]
Updated. Open question is about the extracted test methods.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java (hunk 5 above)
Comment (vvcephei) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: Ah, this is unfortunate, but a good enough justification for me.

Comment (vvcephei) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: I guess the other option would be to disable the long-method checkstyle for tests. this might be reasonable, since test code is often like this (long sequence of operations and assertions) without harming readability.

As opposed to application code, where the methods are full of logic and conditionals and long methods seriously hinder comprehensibility.

I'm curious what @ijuma thinks about potentially making this exclusion for the whole project?

File: streams/src/test/java/org/apache/kafka/test/MockProcessor.java (hunk 8 above)
Comment (vvcephei) [reply streams/src/test/java/org/apache/kafka/test/MockProcessor.java:29]: Sorry, my question was ambiguous. I'm wondering if, since this is a test artifact, we can consider it an internal API and therefore just make the method non-public.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java (hunk 5 above)
Comment (ijuma) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: Yeah, I'm not opposed to that. Another option is to use inline exclusions.

I'd also say that having long test methods is not great and extracting verify methods is not a bad plan. However, having numbers in the name is bad. Typically one can group assertions in a way that makes sense.

Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: So what is the conclusion on that? I am fine any way (merge as-is, remove the checkstyle, rename those method (suggestions for better names), "Typically one can group assertions in a way that makes sense" -- not sure how to apply this to this test...)

Comment (vvcephei) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: Thanks everyone. FWIW, I think my preference is still for putting in a checkstyle exclusion for tests.

But, I'm also fine with merging as-is, if that's your preference @mjsax .

I agree that in general, we can group assertions, but I think in this case it doesn't work out that way.

File: streams/src/test/java/org/apache/kafka/test/MockProcessor.java (hunk 8 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/test/MockProcessor.java:29]: I guess we could, but if one write a test and uses auto-completion it might be better to keep the public -- otherwise, they won't show up.

**Comment by mjsax** [comment]
Rebased. Will merge as-is as no other option seems to be better.

**Comment by mjsax** [comment]
`kafka.api.CustomQuotaCallbackTest.testCustomQuotaCallback` failed.

Retest this please.

**Comment by guozhangwang** [comment]
I'd suggest we add an exception to allow inlines for the long unit test cases. Otherwise I'm +1 on this PR.

Please feel free to merge afterwards.

**Comment by mjsax** [comment]
Updated this:
 - inlined methods
 - added method length checkstyle exception
 - replaced double-brace initialization

**Comment by mjsax** [comment]
> Execution failed for task ':core:compileTestScala'

Not sure how this could have happened. Java8 build passed.

Retest this please.

**Comment by ijuma** [comment]
I fixed a build issue that only affected Java 11 a few hours ago. Maybe you need to rebase to include that.
